packages/node-core/src/utils/graphql.ts

```diff
@@ -140,14 +141,35 @@ export function getColumnOption(
   return columnOption;
 }
 
-function processGetJson(data: any): any {
-  return JSON.parse(JSON.stringify(data), (key, value) => {
-    // regex to check if the value is a bigint string
-    if (typeof value === 'string' && /^-?\d+n$/.test(value)) {
-      return BigInt(value.slice(0, -1));
+/***
+ * Process nested json get, output is same as input value type
+ * @param field
+ * @param value
+ */
+function processGetJson(field: GraphQLEntityField | GraphQLJsonFieldType, value: any): any {
+  // bigIntFields and nestJsonFields from this level in the entity/json
+  const bigIntFields = field.jsonInterface?.fields.filter((f) => f.type === 'BigInt');
+  const nestJsonFields = field.jsonInterface?.fields.filter((f) => f.jsonInterface);
+  const processBigIntFields = (value: any) => {
+    if (bigIntFields && bigIntFields.length) {
+      for (const bigIntField of bigIntFields) {
+        // If null is passed, we should not convert it to BigInt
+        if (value[bigIntField.name] !== undefined && value[bigIntField.name] !== null) {
```

Comment:
Do you think its worth keeping the regex here? Currently hex encoded bigints would cause an error?

Reply:
```
> BigInt("0x05758a60541cc3fa115e")
25780072533029765648734n

```

Hex is working, i think we can leave as is